src/modules/keyboardmanager/common/Helpers.cpp

```diff
@@ -208,30 +208,44 @@ namespace KeyboardManagerHelper
     // Function to return the executable name of the application in focus
     std::wstring GetCurrentApplication(bool keepPath)
     {
-        HWND current_window_handle = GetFocusWindowHandle();
-        DWORD process_id;
-        DWORD nSize = MAX_PATH;
-        WCHAR buffer[MAX_PATH] = { 0 };
-
-        // Get process ID of the focus window
-        DWORD thread_id = GetWindowThreadProcessId(current_window_handle, &process_id);
-        HANDLE hProc = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, process_id);
-
-        // Get full path of the executable
-        bool res = QueryFullProcessImageName(hProc, 0, buffer, &nSize);
+        HWND current_window_handle = GetForegroundWindow();
         std::wstring process_name;
-        CloseHandle(hProc);
 
-        process_name = buffer;
-        if (res)
+        if (current_window_handle != nullptr)
         {
-            PathStripPath(buffer);
+            std::wstring process_path = get_process_path(current_window_handle);
+            process_name = process_path;
+            
+            // Get process name from path
+            PathStripPath(&process_path[0]);
 
+            // Remove elements after null character
+            process_path.erase(std::find(process_path.begin(), process_path.end(), L'\0'), process_path.end());

```

Comment:
This is required because `PathStripPath` is actually designed for `char*` and `wchar*` and not the C++11 `std::string` and `std::wstring`. If the path was 20 characters long (i.e. `process_path` was a 20 size `wstring`), then after using `PathStripPath`, the wstring size remains the same, however it would have many null characters at the end. This causes the basic std::string comparison to fail, as L"Abcd" == L"Abcd\0\0\0" returns false.